addons/cookoff/XEH_postInit.sqf

```diff
@@ -108,3 +108,40 @@ if (isServer) then {
         } forEach allCurators;
     }] call CBA_fnc_addEventHandler;
 };
+
+if (hasInterface) then {
+    [QGVAR(playCookoffSound), {
+        params ["_obj", "_sound", "_maxDis"];
+        if (isNull _obj) exitWith {};
+        private _distance = _obj distance (positionCameraToWorld [0,0,0]);
+        if (_distance > _maxDis) exitWith {};
+        _distance = _distance + (random (100) - 50);
+        private _dis = switch (true) do {
+            case (_distance <= 225): {
+                "close"
+            };
+            case (_distance > 225 && _distance < 852): {
+                "mid"
+            };
+            case (_distance >= 852): {
+                "far"
+            };
+            default {
+                "close"
+            };
+        };
+        _sound = format [QGVAR(%1_%2_%3), _sound, _dis, (floor random 3) + 1];
+        // Delay sound after Rule of SOS.
+        [{
+            params ["_obj", "_sound"];
+
+            private _soundSource = "#particlesource" createVehicleLocal [0,0,0]; // We maybe should switch to a pool of objects that we dont need to create every time a sound plays a new source
+            _soundSource setPos (getPos _obj);
+            _soundSource say3D [_sound, 3000, 0.9 + (random 0.2)];
+            [{
+                params ["_soundObj"];
+                deleteVehicle _soundObj;
+            }, _soundSource, 2] call CBA_fnc_waitAndExecute;
```

Comment:
say3D don't handle that at all. say3D is a direct command. to use SoS engine side you would need to use SoundSets or SoundShader. what means you need to use Ammo or Weapon Firing. else you dont have any SoS